File: TrueVote.Services/Database/Kandidat.cs
using System;
using System.Collections.Generic;

namespace TrueVote.Services.Database;

public partial class Kandidat
{
    public int Id { get; set; }

    public string Ime { get; set; } = null!;

    public string Prezime { get; set; } = null!;

    public int? StrankaId { get; set; }

    public int IzborId { get; set; }

    public byte[]? Slika { get; set; }

    public virtual ICollection<Glas> Glas { get; set; } = new List<Glas>();

    public virtual Izbor Izbor { get; set; } = null!;

    public virtual Stranka? Stranka { get; set; }
}

File: UI/truevote_mobile/lib/models/kandidat.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:truevote_mobile/models/izbor.dart';
import 'package:truevote_mobile/models/stranka.dart';

part 'kandidat.g.dart';

@JsonSerializable()
class Kandidat {
  int id;
  String? ime;
  String? prezime;
  int? strankaId;
  int? izborId;
  String? slika;
  Izbor? izbor;
  Stranka? stranka;

  Kandidat({
    required this.id,
    this.ime,
    this.prezime,
    this.strankaId,
    this.izborId,
    this.slika,
    this.izbor,
    this.stranka,
  });

  factory Kandidat.fromJson(Map<String, dynamic> json) =>
      _$KandidatFromJson(json);

  Map<String, dynamic> toJson() => _$KandidatToJson(this);
}
